File: demos/test-py/isSubsequence.py
def is_subsequence(s, t):
    if not len(s):
        return True
    ps = 0
    pt = 0
    while pt < len(t):
        if s[ps] == t[pt]:
            ps += 1
            pt += 1
            if ps >= len(s):
                return True
        else:
            pt += 1
    return False

File: demos/test-ts/src/algorithms/leetcode/isSubsequence.ts
export function isSubsequence<T>(s: T[], t: T[]): boolean {
  if (!s.length) return true
  let ps = 0
  let pt = 0
  while (pt < t.length) {
    if (s[ps] === t[pt]) {
      ps++
      pt++
      if (ps >= s.length) return true
    } else {
      pt++
    }
  }
  return false
}
